domain/src/SharePooling/SharePoolingId.php

```diff
@@ -5,7 +5,14 @@
 namespace Domain\SharePooling;
 
 use Domain\AggregateRootId;
+use Ramsey\Uuid\Uuid;
 
 final class SharePoolingId extends AggregateRootId
 {
+    private const NAMESPACE = '94ff3977-46f5-4efe-8a89-e474d375232f';
+
+    public static function fromTokenSymbol(string $tokenSymbol): static
+    {
+        return self::fromString(Uuid::uuid5(self::NAMESPACE, trim(strtoupper($tokenSymbol)))->toString());
+    }
```

Comment:
ℹ️ This allows for the generation of a UUID which will always be the same, based on the token's symbol (e.g. BTC)